samlsp/middleware.go

```diff
@@ -90,9 +84,30 @@ func (m *Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	http.NotFoundHandler().ServeHTTP(w, r)
 }
 
+func (m *Middleware) ServeMetadata(w http.ResponseWriter, r *http.Request) {
+	buf, _ := xml.MarshalIndent(m.ServiceProvider.Metadata(), "", "  ")
+	w.Header().Set("Content-Type", "application/samlmetadata+xml")
+	w.Write(buf)
+}
+
+func (m *Middleware) GetAssertion(r *http.Request) (*saml.Assertion, error) {
+	r.ParseForm()
+	assertion, err := m.ServiceProvider.ParseResponse(r, m.getPossibleRequestIDs(r))
+	if err != nil {
+		if parseErr, ok := err.(*saml.InvalidResponseError); ok {
+			m.ServiceProvider.Logger.Printf("RESPONSE: ===\n%s\n===\nNOW: %s\nERROR: %s",
```

Comment:
crewjam saml error logs would be polluting SN logs, which is needed to be standardized later